Create deterministic wasm builds and publish release on tag

Rubberbandits opened the pull request.

Commit:
BIT-353: Use srtool-actions to build deterministic wasm and upload to release

Commit:
BIT-353: Fix github.workspace var and remove macos targets in publish workflow

Commit:
BIT-353: Remove duplicate cache utilization in workflow

Commit:
BIT-353: Clarify release creation workflow name

Commit:
BIT-353: Fix package name in srtools and hardcode chainspec chain arg

Commit:
BIT-353: Switch rustup toolchain from stable to nightly

Commit:
BIT-353: Use toolchain file in publish-tag workflow

Commit:
BIT-353: Remove --tests flag from testing step and use release

This will also prevent duplicate compilation, hopefully speeding up workflows

Commit:
BIT-353: Fix misconfiguration of actions-rs/toolchain for toolchain file

Commit:
BIT-353: Clarify job name for publishing

Commit:
BIT-353: Fix actions-rs/toolchain configuration

Upon further research, this action supports an old format of toolchain spec named `rust-toolchain` which is distinct from `rust-toolchain.toml`

eduardogr commented on .github/workflows/publish-tag.yml:
```diff
@@ -27,49 +27,45 @@ env:
 
         include:
           - os: ubuntu-latest
-          - os: macos-latest
```
can't we build for both ??? and include them all in the release?

Rubberbandits replied:
> can't we build for both ??? and include them all in the release?

Currently macos builds fail due to trying to install deps with apt. We should be using substrate's install script instead of manually installing deps as it will detect platform and install properly for us.

Commit:
BIT-353: Use --tests in testing as some tests in --release dont panic

Commit:
BIT-353: Add wasm32 target to toolchain

Commit:
BIT-353: Remove useless steps in check-rust

Commit:
BIT-353: Add toolchain override for working dir

Commit:
BIT-353: Remove linux target from workflow

Because the rust-target variable is defined in the matrix of the workflow, srtool attempts to build a wasm bin without wasm32 target being valid in one of the workflow instances.
This means that when the linux-gnu target workflow panics, it cancels the wasm target workflow.

Commit:
BIT-353: Update toolchain version and fix rust cache

Commit:
BIT-353: Enforce target and toolchain in srtool build, fix cache

Commit:
BIT-353: Use correct env vars for toolchain and build target

Commit:
BIT-353: Readd TARGET env var

Commit:
BIT-353: Remove CARGO_BUILD_TARGET env var

Commit:
BIT-353: Use +nightly flag in cargo commands, set TARGET env var

Commit:
BIT-353: Use self-hosted runner for build generation

Commit:
BIT-353: Use brew in dep install as self-hosted runner is macOS

Commit:
BIT-353: Update brew install deps

shibshib commented:
Quick note. I think the new nightlies are fixed and cargo should recompile, perhaps let's pick a newer version of nightly?

eduardogr committed:
BIT-353: BIT-353 adding workflow_dispatch for publish tag github action. This allow us to trigger the workflow manually through the Github UI

eduardogr committed:
BIT-353: BIT-353 trigger publish tag workflow for development branch pushes

eduardogr committed:
BIT-353: BIT-353 trigger publish tag workflow for development branch pushes

eduardogr committed:
BIT-353: BIT-353 github-action; publish-tag; use ubuntu-latest

eduardogr committed:
BIT-353: BIT-353 github-action; publish-tag; use ubuntu-latest

eduardogr committed:
BIT-353: BIT-353 github-action; publish-tag; use ubuntu-latest

eduardogr committed:
BIT-353: BIT-353 github-action; publish-tag; use ubuntu-latest

eduardogr committed:
BIT-353: BIT-353 github-action; publish-tag; Use self hosted and exeucting script for dependencies installation

eduardogr committed:
BIT-353: BIT-353 fixing scrips ./scrips/install-deps.sh

eduardogr committed:
BIT-353: BIT-353 fixing scrips ./scrips/install-deps.sh

eduardogr committed:
BIT-353: BIT-353 fixing scrips ./scrips/install-deps.sh

eduardogr committed:
BIT-353: BIT-353 fixing scrips caching for publish-tag.yml workflow

eduardogr committed:
BIT-353: BIT-353 fixing publish-tag.yaml caching

eduardogr committed:
BIT-353: BIT-353 improving publish-tag.yml caching

eduardogr committed:
BIT-353: BIT-353 improving publish-tag.yml caching

eduardogr committed:
BIT-353: BIT-353 adding profile ephemeral build and use opt-level 2 for test profile, using opt-level 3 for release profile

eduardogr committed:
BIT-353: BIT-353 adding profile ephemeral build and use opt-level 2 for test profile, using opt-level 3 for release profile

eduardogr committed:
BIT-353: BIT-353 adding profile ephemeral build and use opt-level 2 for test profile, using opt-level 3 for release profile

eduardogr committed:
BIT-353: BIT-353 upgrade srtool-actions to 0.8.0

eduardogr committed:
BIT-353: fixing publish-tag.yml workflow

eduardogr committed:
BIT-353: BIT-353 debugging github action

eduardogr committed:
BIT-353: BIT-353 +nightly is not a build opt for cargo build. That was proviking the gh action job to fail

eduardogr committed:
BIT-353: BIT-353 trying to fix gh action 'publish-tag' removing option 'runtime_dir: runtime'

eduardogr committed:
BIT-353: BIT-353 trying to fix gh action 'publish-tag'

eduardogr committed:
BIT-353: BIT-353 removing extra 'cargo build'

eduardogr committed:
BIT-353: BIT-353 removing extra step for debugging 'whoami'

eduardogr committed:
BIT-353: BIT-353 trying to fix gh action 'publish-tag' removing option 'runtime_dir: runtime'

eduardogr committed:
BIT-353: BIT-353 UNDO -> trying to fix gh action 'publish-tag' by renaming runtime dir to node-subtensor-runtime

eduardogr committed:
BIT-353: BIT-353 trying to fix gh action 'publish-tag' adding package attribute

eduardogr committed:
BIT-353: BIT-353 trying to fix gh action 'publish-tag' adding package attribute

eduardogr committed:
BIT-353: BIT-353 change cargo profiles a bit

eduardogr committed:
BIT-353: BIT-353 change cargo profiles a bit

eduardogr committed:
BIT-353: BIT-353 change cargo profiles a bit

eduardogr committed:
BIT-353: BIT-353 change cargo profiles a bit

eduardogr committed:
BIT-353: BIT-353 adding all test form pallets/subtensor/test/*.rs into a new directory within pallets/subtensor/test/integration/

eduardogr committed:
BIT-353: BIT-353 change cargo profiles a bit

eduardogr committed:
BIT-353: BIT-353 adding custom image to run srtool

eduardogr committed:
BIT-353: BIT-353 adding custom image to run srtool

eduardogr committed:
BIT-353: BIT-353 adding custom image to run srtool

eduardogr committed:
BIT-353: BIT-353 adding custom image to run srtool

eduardogr committed:
BIT-353: BIT-353 adding custom image to run srtool

eduardogr committed:
BIT-353: BIT-353 adding custom image to run srtool

eduardogr committed:
BIT-353: BIT-353 adding custom image to run srtool
